karapace/kafka_rest_apis/auth_utils.py

```diff
@@ -0,0 +1,136 @@
+"""
+Copyright (c) 2023 Aiven Ltd
+See LICENSE for details
+"""
+from __future__ import annotations
+
+from aiokafka.abc import AbstractTokenProvider as AbstractTokenProviderAsync
+from http import HTTPStatus
+from kafka.oauth.abstract import AbstractTokenProvider
+from karapace.config import Config
+from karapace.rapu import HTTPResponse, JSON_CONTENT_TYPE
+from typing import NoReturn, TypedDict
+
+import aiohttp
+import dataclasses
+import enum
+
+
+@enum.unique
+class TokenType(enum.Enum):
+    BASIC = "Basic"
+    BEARER = "Bearer"
+
+
+def raise_unauthorized() -> NoReturn:
+    raise HTTPResponse(
+        body='{"message": "Unauthorized"}',
+        status=HTTPStatus.UNAUTHORIZED,
+        content_type=JSON_CONTENT_TYPE,
+        headers={"WWW-Authenticate": 'Basic realm="Karapace REST Proxy"'},
```

Comment:
I think the challenge is also dependant of the auth scheme, but I am not sure what to respond as this is called when authorization header is missing and actual mechanisms are defined at Kafka server.

Reply:
Yeah, I wasn't sure either, that's why I left it as-is. It made perfect sense while we only accepted basic auth. In theory we could have multiple of the same header, but `aiohttp` doesn't let it happen.